src/main/java/nus/climods/logic/parser/ParserUtil.java

```diff
@@ -73,7 +76,31 @@ public static Optional<String> parseModuleCode(String input) {
         return Optional.of(input);
     }
 
+    /**
+     * Parse user flag according to user input
+     * @param input Input string from user containing --user. Case insensitive
+     * @return Empty optional if input string does not contain --user, else Optional of --user flag
+     */
+    public static Optional<Boolean> parseUserFlag(String input) {
+        if (!USER_FLAG_PATTERN.matcher(input.trim()).find()) {
+            return Optional.empty();
+        }
+        return Optional.of(Boolean.TRUE);
+    }
 
+    /**
+     * Parse faculty code according to user input
+     * @param input Input string from user containing faculty code
+     * @return
+     */
+    public static Optional<String> parseFacultyCode(String input) {
+        if (input.length() == 0) {
+            return Optional.empty();
+        }
+        input = !FACULTY_CODE_PATTERN.matcher(input.trim()).find() ? FAULTY_FACULTY_CODE : input;
```

Comment:
Why do we need to return a `FAULTY_FACULTY_CODE` here? instead of just `Optional.Empty`? Is it being used for something?

Reply:
The behaviour for Optional.Empty is a bit unexpected because Predicate will evaluate True for every module.
This is ok if we just did ls without any module code.
But if we have an invalid string of invalid length (E.g. ls csssssssss) then it wouldn't be captured by the regex and will be treated as an Optional.Empty --> all modules will be returned. 
I think the correct behaviour here should be to return 0 modules, which is why i returned a FAULTY_FACULTY_CODE instead

Reply:
this feels a lil hacky 😅 hmmm i guess we can merge it first for the interest of time but add a `TODO` there to make sure we get back to this